### DISCUSSION TITLE
The call of training_step and validation_step .etc.

### DISCUSSION THREAD

**Discussion by MrChenFeng** [Lightning-AI/pytorch-lightning#10070] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Anyone help me where did these funcs been called in the core parts? I expect it to be called in the loop instance of trainer however not. Quite confused about this.

**Comment by rohitgr7** [comment, marked as answer]
depends upon the configuration but in a general case here: https://github.com/PyTorchLightning/pytorch-lightning/blob/454e93bacecb4f1d6ebc9ebd3bc11b26723ddbec/pytorch_lightning/plugins/training_type/training_type_plugin.py#L217-L230

call sequence is `loop -> accelerators -> training_type_plugin -> actual call`
